apps/meteor/ee/server/local-services/omnichannel/OmniEEService.ts

```diff
@@ -0,0 +1,175 @@
+
+	private logger = schedulerLogger;
+
+	private systemUser: IUser;
+
+	constructor() {
+		super();
+		this.logger.debug('OmniEEService started');
+	}
+
+	async monitorOnHoldRoomForAutoClose(roomId: string, timeout: number, comment: string): Promise<void> {
+		this.logger.debug(`monitorOnHoldRoomForAutoClose: ${roomId}`);
+		await this.cancelMonitorOnHoldRoomForAutoClose(roomId);
+
+		const when = moment(new Date()).add(timeout, 's').toDate();
+		const job = await OmniJobSchedularService.scheduleJobAt<OmniOnHoldJobData>(OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT, when, {
+			roomId,
+			comment,
+		});
+
+		this.logger.debug(`Scheduled job: ${OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT} for room ${roomId} at ${job.attrs.nextRunAt}`);
+	}
+
+	async cancelMonitorOnHoldRoomForAutoClose(roomId: string): Promise<void> {
+		this.logger.debug(`cancelMonitorOnHoldRoomForAutoClose: ${roomId}`);
+
+		const totalCancelledJobs = await OmniJobSchedularService.cancelJobByRoomId(OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT, roomId);
+
+		this.logger.debug(`Unscheduled ${OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT} for room ${roomId} (${totalCancelledJobs} jobs cancelled)`);
+	}
+
+	async autoCloseOnHoldChat(job: Job): Promise<void> {
+		const data = job.attrs.data as OmniOnHoldJobData;
+
+		this.logger.debug(`Executing job: ${OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT} for room ${data.roomId}`);
+
+		const { roomId, comment } = data;
+		const [room, user] = await Promise.all([LivechatRooms.findOneById(roomId), this.getSystemUser()]);
+		if (!room) {
+			this.logger.error(`Could not find room ${roomId}`);
+		}
+		const payload = {
+			user,
+			room,
+			comment,
+			options: {},
+			visitor: undefined,
+		};
+		Livechat.closeRoom(payload);
+
+		this.logger.debug(`Executed ${OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT} for room ${roomId}`);
+	}
+
+	async monitorUnansweredRoomForAutoTransfer(roomId: string, timeout: number): Promise<void> {
+		this.logger.debug(`monitorUnansweredRoomForAutoTransfer: ${roomId}`);
+		await this.cancelMonitorUnansweredRoomForAutoTransfer(roomId);
+
+		const when = moment(new Date()).add(timeout, 's').toDate();
+
+		const [job] = await Promise.all([
+			OmniJobSchedularService.scheduleJobAt<OmniAutoTransferUnansweredChatJobData>(OMNI_JOB_NAME.AUTO_TRANSFER_UNANSWERED_CHAT, when, {
+				roomId,
+			} as OmniAutoTransferUnansweredChatJobData),
+			LivechatRooms.setAutoTransferOngoingById(roomId),
+		]);
+
+		this.logger.debug(`Scheduled job: ${OMNI_JOB_NAME.AUTO_TRANSFER_UNANSWERED_CHAT} for room ${roomId} at ${job.attrs.nextRunAt}`);
+	}
+
+	async cancelMonitorUnansweredRoomForAutoTransfer(roomId: string): Promise<void> {
+		this.logger.debug(`cancelMonitorUnansweredRoomForAutoTransfer: ${roomId}`);
+
+		const [, totalCancelledJobs] = await Promise.all([
+			LivechatRooms.unsetAutoTransferOngoingById(roomId),
+			OmniJobSchedularService.cancelJobByRoomId(OMNI_JOB_NAME.AUTO_TRANSFER_UNANSWERED_CHAT, roomId),
+		]);
+
+		this.logger.debug(
```

Comment:
Some of those logs should be on the "OmniJobSchedulerService"

Reply:
Makes sense :+1: I've moved all the relevant scheduling & un-scheduling logs inside the scheduling service itself